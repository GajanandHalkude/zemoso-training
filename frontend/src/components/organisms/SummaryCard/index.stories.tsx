import { ComponentStory, ComponentMeta } from '@storybook/react'
import SummaryCard from "./index";
import React from 'react'

export default {
  title: "Organisms/SummaryCard",
  component: SummaryCard,
} as ComponentMeta<typeof SummaryCard>;

const Template: ComponentStory<typeof SummaryCard> = (args) => (
  <SummaryCard {...args} />
);

export const SellSummaryCard = Template.bind({})

SellSummaryCard.args = {
  type: "sell",
  btcValue: 0.023451,
};

export const BuySummaryCard = Template.bind({});

BuySummaryCard.args = {
  type: "buy",
  btcValue: 0.023451,
};