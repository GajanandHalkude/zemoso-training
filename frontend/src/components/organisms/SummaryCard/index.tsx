import { styled, Box, Grid, Stack } from "@mui/material";
import React from 'react'
import TypographyComponent from '../../atoms/typography'
import { dollarValue, formatCurrency } from '../../../constants'
import theme from '../../../theme'
import IconComponent from '../../atoms/icon'
import dottedLine from "../../../../public/assets/icons/orderSummary.svg";
import ButtonComponent from '../../atoms/button'
import PaymentSummaryStepper from "../../molecules/PaymentSummaryStepper";


interface SummaryCardProps {
  type: "buy" | "sell";
  btcValue: number;
  price?: number;
  onClick: (arg: any) => void;
  amount?: number;
}
const StyledBox = styled(Box)({
  width: "527px",
  border: `1px solid ${theme.palette.greyColors.grey100}`,
  borderRadius: "4px",
  background: theme.palette.structural.main,
});

const TypographyGrid = styled(Grid)({
  paddingTop: "28px",
  paddingBottom: "32px",
  alignItems: "center",
  justifyContent: "center",
  margin: "0px",
  width: "100%",
  borderBottom: `1px solid ${theme.palette.greyColors.grey100}`,
});

const StepperBox = styled(Box)({
  padding: "24px 0 24px 24px",
  borderBottom: `1px solid ${theme.palette.greyColors.grey100}`,
});

const TotalBox = styled(Stack)({
  padding: "24px 0 70px 24px",
});

const renderOrderSummary = (
  leftValue: string,
  rightValue: string,
  width: string,
  bold: boolean
) => {
  return (
    <Stack direction="row" spacing={-0.5}>
      <TypographyComponent
        variant={bold ? "body1" : 'overline'}
        text={leftValue}
        color={theme.palette.textColor.highEmphasis}
        sx={{textTransform:'none'}}
      />
      <IconComponent
        src={dottedLine}
        width={width}
        height={"4px"}
        padding={bold ? '0 0 1px 0' : '0 0 4px 0'}
      />
      <TypographyComponent
        variant={bold ? "body1" : 'overline'}
        text={`${rightValue}`}
        color={theme.palette.textColor.highEmphasis}
        display="flex"
        justifyContent={"end"}
      />
    </Stack>
  );
};

const StyledButtonComponent = styled(ButtonComponent)({
  width: "479px",
  height: "42px",
});

const typographyStyle = {
  color: theme.palette.textColor.mediumEmphasis,
};

const SummaryCard: React.FC<SummaryCardProps> = ({
  type,
  btcValue,
  price,
  onClick,
  amount,
}) => {
  price = price ? price : dollarValue;
  amount = amount ? amount : btcValue * price;
  const total = type === "buy" ? amount + 1000 : amount - 1000;
  return (
    <StyledBox data-testid="summary-card">
      <TypographyGrid container spacing="12px" direction={"column"}>
        <Grid item xs={12}>
          <TypographyComponent
            variant="caption1"
            text={`You are ${type}ing`}
            style={typographyStyle}
          />
        </Grid>
        <Grid item xs={12}>
          <TypographyComponent variant="h6" text={`${btcValue} BTC`} />
        </Grid>
        <Grid item xs={12}>
          <TypographyComponent
            variant="caption1"
            text={`1BTC = ${formatCurrency.format(price)}`}
            style={typographyStyle}
          />
        </Grid>
      </TypographyGrid>

      <StepperBox>
        <PaymentSummaryStepper />
      </StepperBox>

      <TotalBox spacing={2} direction={"column"}>
        {renderOrderSummary(
          `${btcValue} BTC`,
          formatCurrency.format(amount),
          "330px",
          false
        )}
        {renderOrderSummary("transaction fee", "$1,000.00", "333px", false)}
        {renderOrderSummary(
          "Total",
          formatCurrency.format(total),
          "330px",
          true
        )}
        <StyledButtonComponent
          variant="contained"
          text={type === "buy" ? "BUY NOW" : "SELL NOW"}
          backgroundColor={
            type === "buy"
              ? theme.palette.primary.main
              : theme.palette.primary.warning300
          }
          onClick={() => {
            onClick(btcValue);
          }}
        />
      </TotalBox>
    </StyledBox>
  );
};

export default SummaryCard;
