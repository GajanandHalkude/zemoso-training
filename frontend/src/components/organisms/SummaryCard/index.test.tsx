import renderer from 'react-test-renderer';
import SummaryCard from './index';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom'
import React from 'react';

const handleChange = jest.fn()
it('renders buy summary card correctly', () => {
    const tree = renderer
      .create(
        <SummaryCard type="buy" btcValue={0.023451} onClick={handleChange} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders sell summary card correctly', () => {
    const tree = renderer
      .create(
        <SummaryCard type="sell" btcValue={0.023451} onClick={handleChange} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
});

describe("Organisms/SummaryCard", () => {
  test("Renders sell card correctly", () => {
    render(
      <SummaryCard type="sell" btcValue={0.023451} onClick={handleChange} />
    );
    expect(screen.getByRole("heading")).toHaveTextContent("0.023451 BTC");
    expect(screen.getAllByRole("img")).toHaveLength(8);
    expect(screen.getByRole("button")).toHaveTextContent("SELL NOW");
  });

  test("Renders buy card correctly", () => {
    render(
      <SummaryCard type="buy" btcValue={0.023451} onClick={handleChange} />
    );
    expect(screen.getByRole("heading")).toHaveTextContent("0.023451 BTC");
    expect(screen.getAllByRole("img")).toHaveLength(8);
    expect(screen.getByRole("button")).toHaveTextContent("BUY NOW");
  });
});