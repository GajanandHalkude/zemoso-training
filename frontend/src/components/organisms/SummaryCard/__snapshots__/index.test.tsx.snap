// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders buy summary card correctly 1`] = `
<div
  className="MuiBox-root css-ufn7cb"
  data-testid="summary-card"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-tc7jh2-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
        data-testid="typography-component"
        style={
          {
            "color": "#7D7D89",
          }
        }
      >
        You are buying
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        data-testid="typography-component"
      >
        0.023451 BTC
      </h6>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
        data-testid="typography-component"
        style={
          {
            "color": "#7D7D89",
          }
        }
      >
        1BTC = $3,406,069.54
      </p>
    </div>
  </div>
  <div
    className="MuiBox-root css-iwz9lp"
  >
    <div>
      <div
        className="MuiBox-root css-15budcu"
        data-testid="summary-stepper"
      >
        <div
          className="MuiBox-root css-1oyjow8"
        >
          <div
            className="MuiBox-root css-1j5om8t"
          >
            <div>
              <img
                alt="icon"
                data-testid="iconComponent"
                height="32px"
                src=""
                style={
                  {
                    "padding": "5px 6px",
                  }
                }
                width="32px"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-15budcu"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
              data-testid="typography-component"
              style={
                {
                  "color": "#7D7D89",
                }
              }
            >
              Payment method
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              data-testid="typography-component"
              style={
                {
                  "color": "#343446",
                }
              }
            >
              Visa credit ...8845
            </p>
          </div>
        </div>
        <div>
          <img
            alt="icon"
            data-testid="iconComponent"
            height="40px"
            src=""
            style={
              {
                "padding": undefined,
              }
            }
            width="40px"
          />
        </div>
        <div
          className="MuiBox-root css-1oyjow8"
        >
          <div
            className="MuiBox-root css-1j5om8t"
          >
            <div>
              <img
                alt="icon"
                data-testid="iconComponent"
                height="32px"
                src=""
                style={
                  {
                    "padding": "5px 6px",
                  }
                }
                width="32px"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-15budcu"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
              data-testid="typography-component"
              style={
                {
                  "color": "#7D7D89",
                }
              }
            >
              Delivery fees
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              data-testid="typography-component"
              style={
                {
                  "color": "#343446",
                }
              }
            >
              0.001 BTC
            </p>
          </div>
        </div>
        <div>
          <img
            alt="icon"
            data-testid="iconComponent"
            height="40px"
            src=""
            style={
              {
                "padding": undefined,
              }
            }
            width="40px"
          />
        </div>
        <div
          className="MuiBox-root css-1oyjow8"
        >
          <div
            className="MuiBox-root css-1j5om8t"
          >
            <div>
              <img
                alt="icon"
                data-testid="iconComponent"
                height="32px"
                src=""
                style={
                  {
                    "padding": "5px 6px",
                  }
                }
                width="32px"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-15budcu"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
              data-testid="typography-component"
              style={
                {
                  "color": "#7D7D89",
                }
              }
            >
              Deposit to
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              data-testid="typography-component"
              style={
                {
                  "color": "#343446",
                }
              }
            >
              Bitcoin wallet
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiStack-root css-dlxwab-MuiStack-root"
  >
    <div
      className="MuiStack-root css-6hgg78-MuiStack-root"
    >
      <span
        className="MuiTypography-root MuiTypography-overline css-1wrpxv1-MuiTypography-root"
        data-testid="typography-component"
      >
        0.023451 BTC
      </span>
      <div>
        <img
          alt="icon"
          data-testid="iconComponent"
          height="4px"
          src=""
          style={
            {
              "padding": "0 0 4px 0",
            }
          }
          width="330px"
        />
      </div>
      <span
        className="MuiTypography-root MuiTypography-overline css-34k6ax-MuiTypography-root"
        data-testid="typography-component"
      >
        $79,875.74
      </span>
    </div>
    <div
      className="MuiStack-root css-6hgg78-MuiStack-root"
    >
      <span
        className="MuiTypography-root MuiTypography-overline css-1wrpxv1-MuiTypography-root"
        data-testid="typography-component"
      >
        transaction fee
      </span>
      <div>
        <img
          alt="icon"
          data-testid="iconComponent"
          height="4px"
          src=""
          style={
            {
              "padding": "0 0 4px 0",
            }
          }
          width="333px"
        />
      </div>
      <span
        className="MuiTypography-root MuiTypography-overline css-34k6ax-MuiTypography-root"
        data-testid="typography-component"
      >
        $1,000.00
      </span>
    </div>
    <div
      className="MuiStack-root css-6hgg78-MuiStack-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1sjk0ek-MuiTypography-root"
        data-testid="typography-component"
      >
        Total
      </p>
      <div>
        <img
          alt="icon"
          data-testid="iconComponent"
          height="4px"
          src=""
          style={
            {
              "padding": "0 0 1px 0",
            }
          }
          width="370px"
        />
      </div>
      <p
        className="MuiTypography-root MuiTypography-body1 css-179i82-MuiTypography-root"
        data-testid="typography-component"
      >
        $80,875.74
      </p>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-ib6dj5-MuiButtonBase-root-MuiButton-root"
      data-testid="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      BUY NOW
    </button>
  </div>
</div>
`;

exports[`renders sell summary card correctly 1`] = `
<div
  className="MuiBox-root css-ufn7cb"
  data-testid="summary-card"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-tc7jh2-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
        data-testid="typography-component"
        style={
          {
            "color": "#7D7D89",
          }
        }
      >
        You are selling
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        data-testid="typography-component"
      >
        0.023451 BTC
      </h6>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
        data-testid="typography-component"
        style={
          {
            "color": "#7D7D89",
          }
        }
      >
        1BTC = $3,406,069.54
      </p>
    </div>
  </div>
  <div
    className="MuiBox-root css-iwz9lp"
  >
    <div>
      <div
        className="MuiBox-root css-15budcu"
        data-testid="summary-stepper"
      >
        <div
          className="MuiBox-root css-1oyjow8"
        >
          <div
            className="MuiBox-root css-1j5om8t"
          >
            <div>
              <img
                alt="icon"
                data-testid="iconComponent"
                height="32px"
                src=""
                style={
                  {
                    "padding": "5px 6px",
                  }
                }
                width="32px"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-15budcu"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
              data-testid="typography-component"
              style={
                {
                  "color": "#7D7D89",
                }
              }
            >
              Payment method
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              data-testid="typography-component"
              style={
                {
                  "color": "#343446",
                }
              }
            >
              Visa credit ...8845
            </p>
          </div>
        </div>
        <div>
          <img
            alt="icon"
            data-testid="iconComponent"
            height="40px"
            src=""
            style={
              {
                "padding": undefined,
              }
            }
            width="40px"
          />
        </div>
        <div
          className="MuiBox-root css-1oyjow8"
        >
          <div
            className="MuiBox-root css-1j5om8t"
          >
            <div>
              <img
                alt="icon"
                data-testid="iconComponent"
                height="32px"
                src=""
                style={
                  {
                    "padding": "5px 6px",
                  }
                }
                width="32px"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-15budcu"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
              data-testid="typography-component"
              style={
                {
                  "color": "#7D7D89",
                }
              }
            >
              Delivery fees
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              data-testid="typography-component"
              style={
                {
                  "color": "#343446",
                }
              }
            >
              0.001 BTC
            </p>
          </div>
        </div>
        <div>
          <img
            alt="icon"
            data-testid="iconComponent"
            height="40px"
            src=""
            style={
              {
                "padding": undefined,
              }
            }
            width="40px"
          />
        </div>
        <div
          className="MuiBox-root css-1oyjow8"
        >
          <div
            className="MuiBox-root css-1j5om8t"
          >
            <div>
              <img
                alt="icon"
                data-testid="iconComponent"
                height="32px"
                src=""
                style={
                  {
                    "padding": "5px 6px",
                  }
                }
                width="32px"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-15budcu"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
              data-testid="typography-component"
              style={
                {
                  "color": "#7D7D89",
                }
              }
            >
              Deposit to
            </p>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              data-testid="typography-component"
              style={
                {
                  "color": "#343446",
                }
              }
            >
              Bitcoin wallet
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiStack-root css-dlxwab-MuiStack-root"
  >
    <div
      className="MuiStack-root css-6hgg78-MuiStack-root"
    >
      <span
        className="MuiTypography-root MuiTypography-overline css-1wrpxv1-MuiTypography-root"
        data-testid="typography-component"
      >
        0.023451 BTC
      </span>
      <div>
        <img
          alt="icon"
          data-testid="iconComponent"
          height="4px"
          src=""
          style={
            {
              "padding": "0 0 4px 0",
            }
          }
          width="330px"
        />
      </div>
      <span
        className="MuiTypography-root MuiTypography-overline css-34k6ax-MuiTypography-root"
        data-testid="typography-component"
      >
        $79,875.74
      </span>
    </div>
    <div
      className="MuiStack-root css-6hgg78-MuiStack-root"
    >
      <span
        className="MuiTypography-root MuiTypography-overline css-1wrpxv1-MuiTypography-root"
        data-testid="typography-component"
      >
        transaction fee
      </span>
      <div>
        <img
          alt="icon"
          data-testid="iconComponent"
          height="4px"
          src=""
          style={
            {
              "padding": "0 0 4px 0",
            }
          }
          width="333px"
        />
      </div>
      <span
        className="MuiTypography-root MuiTypography-overline css-34k6ax-MuiTypography-root"
        data-testid="typography-component"
      >
        $1,000.00
      </span>
    </div>
    <div
      className="MuiStack-root css-6hgg78-MuiStack-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1sjk0ek-MuiTypography-root"
        data-testid="typography-component"
      >
        Total
      </p>
      <div>
        <img
          alt="icon"
          data-testid="iconComponent"
          height="4px"
          src=""
          style={
            {
              "padding": "0 0 1px 0",
            }
          }
          width="370px"
        />
      </div>
      <p
        className="MuiTypography-root MuiTypography-body1 css-179i82-MuiTypography-root"
        data-testid="typography-component"
      >
        $78,875.74
      </p>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1k23lp7-MuiButtonBase-root-MuiButton-root"
      data-testid="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      SELL NOW
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
